kore/src/Kore/Builtin/Set.hs

```diff
@@ -499,6 +589,628 @@ isSymbolUnit
     -> Bool
 isSymbolUnit = Builtin.isSymbol "SET.unit"
 
+{- | A normalized representation of a set, with elements separated from
+other set terms, and with concrete elements separated from non-concrete ones
+-}
+-- TODO(virgil): Use this as the internal representation of a set.
```

Comment:
I think the problem you were having with symbol attributes is related to using `listAsTermLike` above. I think if you did use `NormalizedSet` as the internal representation of sets you would not need to use `listAsTermLike`, is that right?

Reply:
Yes, it's about `listAsTermLike` and `normalizedToTerm`.

I didn't try to switch the default representation to NormalizedSet because it seemed to require its own PR. I didn't think about the definedness issues you mentioned, but I guessed that we would need to be careful about it.

To make this PR work, I added symbol attributes to `MetadataTools`, it seemed to be the safest way of solving this problem. If you think this is wrong, the entire symbol attribute code is in its own commit, so it should be easy for me to rollback and do something else.

Reply:
That sounds good to me!